syntax = "proto3";

package ai_call;

option go_package = ".;ai_service";

import "google/protobuf/timestamp.proto";

// The chat service definition.
service AIService {
  // Processes a chat message and returns a response
  rpc Process (Request) returns (Response) {}
}

// The request message containing the user's chat information.
message Request {
  string user_id = 1;  // UUID of the user
  string session_id = 2;  // UUID of the session
  string chat_message = 3;  // The chat message to process
  google.protobuf.Timestamp timestamp = 4;  // Timestamp of the request
}

// The response message containing the server's response.
message Response {
  string response_text = 1;  // The server's response to the chat
  google.protobuf.Timestamp timestamp = 2;  // Timestamp of the response
}